### PR TITLE
S4T6: Parser for getting info about gas shutdowns

### PR DESCRIPTION
**Pull request by DmitriyOsipov**

### CODE REVIEW

File: src/main/java/com/cityelf/utils/GasLoader.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      throw new ParserUnavailableException(ex.getMessage(), ex.getCause());
+    }
+    return loaded;
+  }
+
+  private Elements findNewsBlock() throws ParserUnavailableException {
+    Elements newsBlock = this.loadDocument().select("div.news-content");
+    if (newsBlock.isEmpty()) {
+      throw new ParserUnavailableException("No news block found",
+          new Throwable("Gas page structure was changed"));
```
Comment (parkhomenko) [inline src/main/java/com/cityelf/utils/GasLoader.java]: Why do you need to create a new Throwable here?

File: src/test/java/com/cityelf/utils/ParserGasTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+public class ParserGasTest {
+
+  @Autowired
+  private ParserGas parserGas;
+
+  private String content;
+
+  @Before
+  public void setUp() throws Exception {
+    File file = new File(".\\src\\test\\java\\com\\cityelf\\utils\\gasWebResponseTest.txt");
```
Comment (parkhomenko) [inline src/test/java/com/cityelf/utils/ParserGasTest.java]: You have to move this file into the test\resources folder. This folder is exactly for such files.
Look here how to read it then:
https://www.mkyong.com/java/java-read-a-file-from-resources-folder/

File: src/main/java/com/cityelf/utils/GasForecaster.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+class GasForecaster {
+
+  @Autowired
+  private StreetExtractor streetExtractor;
+
+  @Autowired
+  private NumberExtractor numberExtractor;
+
+
+  void setStreetExtractor(StreetExtractor streetExtractor) {
```
Comment (parkhomenko) [inline src/main/java/com/cityelf/utils/GasForecaster.java]: Why do you have some function without access modifier?

**Review by parkhomenko — CHANGES_REQUESTED** [review]
Fix some minor issues.

File: src/main/java/com/cityelf/utils/GasForecaster.java (hunk 3 above)
Comment (DmitriyOsipov) [reply src/main/java/com/cityelf/utils/GasForecaster.java]: here was two setters which I didn't use, so I removed them.
For some others - functions are package-private and may be used only inside package utils (and are using only in ParserGas class)

File: src/test/java/com/cityelf/utils/ParserGasTest.java (hunk 2 above)
Comment (DmitriyOsipov) [reply src/test/java/com/cityelf/utils/ParserGasTest.java]: Done.
Also I added two tests for start and end dates

File: src/main/java/com/cityelf/utils/GasLoader.java (hunk 1 above)
Comment (DmitriyOsipov) [reply src/main/java/com/cityelf/utils/GasLoader.java]: I added a new exception type: GasPageStructureChangedException. It will be more logical

**Review by parkhomenko — APPROVED** [review]

**Pull request merged**